backend/tests/app/routes/conftest.py

```diff
@@ -0,0 +1,31 @@
+# pylint: disable=missing-module-docstring,missing-function-docstring
+import json
+import os
+from unittest.mock import AsyncMock, Mock, create_autospec
+
+import pytest
+from fastapi import FastAPI
+
+from app.solr.operations import search, search_dep
+
+
+@pytest.fixture
+def mock_post_search(app: FastAPI) -> AsyncMock:
+    mock_search = get_mock("test_search.post.response.json")
+
+    app.dependency_overrides[search_dep] = lambda: mock_search
+    yield mock_search
+    del app.dependency_overrides[search_dep]
+
+
+def get_mock(file: str) -> AsyncMock:
+    mock_json = json.loads(read_file_contents(file))
+    mock_return = Mock()
+    mock_return.is_error = False
+    mock_return.json = Mock(return_value=mock_json)
+    return create_autospec(search, return_value=mock_return)
+
+
+def read_file_contents(file: str):
+    with open(f"{os.path.dirname(__file__)}/{file}", "r", encoding="utf-8") as f:
+        return f.read()
```

Comment:
I would suggest moving `json.loads` inside of this function and calling it `read_file_as_json`.
Additionally I strongly advice using `pathlib` module instead of `os.path` module. `Path` is a much more Object Oriented way of doing things with file paths: (In this short case the code written would totally break on windows (which in our case is probably acceptable, but it is still something good to avoid as a basic rule - rember - not all computer systems use `/` as a separator in file systems)

```
from pathlib import Path
with open(Path(__file__).parent / file, "r", encoding="utf-8") as f:
        return f.read()

```

Also  - encoding is usually understood and default is `"utf-8"` - but being verbose is not wrong.